DNN Platform/Library/Entities/Controllers/IHostController.cs

```diff
@@ -26,46 +26,149 @@
+        /// <param name="key">The setting key string</param>
+        /// <param name="passPhrase">The passPhrase used to decrypt the setting value</param>
+        /// <returns>The setting value as a <see cref="string"/></returns>
         string GetEncryptedString(string key,string passPhrase);
 
+        /// <summary>
+        /// Gets the setting value for a specific key
+        /// </summary>
+        /// <param name="key">The setting key string</param>
+        /// <returns>The setting value as a <see cref="string"/></returns>
         string GetString(string key);
 
+        /// <summary>
+        /// Gets the setting value for a specific key
+        /// </summary>
+        /// <param name="key">The seeting key string</param>
+        /// <param name="defaultValue"></param>
+        /// <returns>Default value returned if the setting is not found</returns>
         string GetString(string key, string defaultValue);
 
+        /// <summary>
+        /// Updates the specified settings
+        /// </summary>
+        /// <param name="settings">The settings to update</param>
         void Update(Dictionary<string, string> settings);
 
+        /// <summary>
+        /// Updates the specified config
+        /// </summary>
+        /// <param name="config">The configuration setting</param>
         void Update(ConfigurationSetting config);
 
+        /// <summary>
+        /// Updates the specified config, ontionally clearing the cache
+        /// </summary>
+        /// <param name="config">The configuaration setting</param>
+        /// <param name="clearCache">If set to <c>true</c>, will clear the cache after updating the setting.</param>
         void Update(ConfigurationSetting config, bool clearCache);
 
+        /// <summary>
+        /// Updates the setting for a specific key
+        /// </summary>
+        /// <param name="key">The setting key string</param>
+        /// <param name="value">The value to update</param>
+        /// <param name="clearCache">If set to <c>true</c>, will clear the cache after updating the setting</param>
         void Update(string key, string value, bool clearCache);
 
+        /// <summary>
+        /// Updates the setting for a specific key
+        /// </summary>
+        /// <param name="key">The setting key string</param>
+        /// <param name="value">The value to update</param>
         void Update(string key, string value);
 
+        /// <summary>
+        /// Takes in a <see cref="string"/> value, encrypts it with a FIPS compliant algorithm and stores
+        /// </summary>
+        /// <param name="key">host settings key</param>
+        /// <param name="value">host settings value</param>
+        /// <param name="passPhrase">pass phrase to allow encryption/decryption</param>
         void UpdateEncryptedString(string key, string value, string passPhrase);
 
+        /// <summary>
+        /// Increments the Client Resource Manager (CRM) version to bust local cache
```

Comment:
Bust local cache? I think we should go with `rebuild the local cache`.

Reply:
I don't know, isn't cache-busting the proper term here? We are not caching, we are just forcing the browser to bust the current cache, I have known this technique by that name, using a unique value in a url to bust the local cache. Google seems to agree with the term.
https://www.google.ca/search?q=cachebusting&oq=cachebusting&aqs=chrome..69i57j0l5.2238j0j7&sourceid=chrome&ie=UTF-8